service/bulk_create.go

```diff
@@ -46,13 +46,13 @@ func BulkAssembly(req m.BulkCreateRequest, tenant *m.Tenant, user *m.User) (*m.B
 		}
 
 		userDao := dao.GetUserDao(&tenant.Id)
-		err = userDao.CreateIfResourceOwnershipActive(userResource)
+		err = userDao.FindOrCreateUserIfResourceOwnershipActive(userResource)
```

Comment:
After the live PR review I now know that this is where the user is being created - we basically just need to move this piece over to the middleware where you are pulling the user id from the header, and create the user record [like we do for tenancy](https://github.com/RedHatInsights/sources-api-go/blob/9a7c9288be53f84717a6337063a481dcf533f615/middleware/tenancy.go#L103-L109).

From here - you'll just instantiate the userResource struct with the id from the current context rather than having to create teh record here.

Reply:
Done here https://github.com/RedHatInsights/sources-api-go/pull/412